docs/koji.md

```diff
@@ -0,0 +1,127 @@
+## Koji integration
+
+The `add_filesystem` and `koji` pre-build plugins and the `koji_promote` exit plugin provide integration with [Koji](https://docs.pagure.org/koji/).
+
+## Pre-build plugins
+
+The `add_filesystem` pre-build plugin provides special handling for images with "FROM koji/image-build". For these images it creates a Koji task to create a installed filesystem archive as [described here](https://github.com/projectatomic/atomic-reactor/blob/master/docs/base_images.md).
+
+The `koji` pre-build plugin creates a yum repofile to allow instructions in the Dockerfile to have access to RPMs available from the appropriate Koji tag.
+
+## Exit plugin
+
+The `koji_promote` exit plugin uses the [Koji Content Generator API](https://docs.pagure.org/koji/content_generators/) to import the built image, and logs generated during the build, into Koji.
+
+When importing a build using the Content Generator API, metadata to describe the build is generated. This follows [the metadata format specification](https://docs.pagure.org/koji/content_generator_metadata/) but in some cases content generators are free to choose how metadata is expressed.
+
+Each build creates a single output archive, in the [Combined Image JSON + Filesystem Changeset format](https://github.com/docker/docker/blob/master/image/spec/v1.2.md#combined-image-json--filesystem-changeset-format).
+
+# Type-specific build metadata
+
+For atomic-reactor container image builds the 'image' type is used, and so type-specific information is placed into the build.extra.image map. Note that this does not use build.extra.typeinfo.image -- the 'typeinfo' element was introduced after the 'image' type started to be used.
+
+Data which is placed here includes:
+
+- build.extra.image.autorebuild (boolean): true if this build was triggered automatically; false otherwise
```

Comment:
Nitpick: perhaps those should be wrapped in backticks: `build.extra.image.autorebuild`?